_source/_posts/2022-08-30-introducing-the-new-okta-mobile-sdks.md

```diff
@@ -0,0 +1,333 @@
+
+```swift
+if let deviceSecret = credential.token.deviceSecret,
+   let idToken = credential.token.idToken
+{
+    try Keychain.Item(account: "ssoSecret",
+                      accessGroup: "com.myapp.shared",
+                      value: deviceSecret).save()
+    try Keychain.Item(account: "ssoToken",
+                      accessGroup: "com.myapp.shared",
+                      value: idToken).save()
+}
+```
+
+Later, retrieving these values can be an equally straightforward operation.
+
+```swift
+if let secretData = try? Keychain.Search(account: "ssoSecret").get(),
+   let secretToken = try? Keychain.Search(account: "ssoToken").get(),
+   let ssoSecret = String(data: secretData, encoding: .utf8),
+   let ssoToken = String(data: secretData, encoding: .utf8)
+{
+    // Exchange the tokens using TokenExchangeFlow
+}
+```
+
+#### Outgoing request authorization
+
+Ultimately, developers use the Okta Mobile SDK in order to securely access resources on a server somewhere. Your users authenticate using their credentials, and your application can use the resulting tokens to perform network requests on their behalf.
+
+Since this is such a common operation, we decided to improve the developer experience by streamlining the process.
+
+```swift
+let url = URL(string: "https://api.example.com/notes")!
+var request = URLRequest(url: url)
+request.httpMethod = "POST"
+request.httpBody = jsonBody
+
+// Add appropriate HTTP authorization headers,
+// automatically refreshing the token if needed.
+await credential.authorize(&request)
+
+let (data, response) = try await URLSession.shared
+    .data(for: request)
+```
+
+When using the Swift Concurrency [Credential.authorize()](https://okta.github.io/okta-mobile-swift/development/authfoundation/documentation/authfoundation/credential/authorize(_:)) async function, this will automatically refresh the credential's access token if it's expired, and add the appropriate HTTP authorization header to the request object. If you're not using Swift Concurrency, the same thing can be accomplished with the [refreshIfNeeded(graceInterval:completion:)](https://okta.github.io/okta-mobile-swift/development/authfoundation/documentation/authfoundation/credential/refreshifneeded(graceinterval:completion:)) and [authorize(request:)](https://okta.github.io/okta-mobile-swift/development/authfoundation/documentation/authfoundation/credential/authorize(request:)) functions.
```

Comment:
```suggestion
When using the Swift Concurrency [`Credential.authorize()`](https://okta.github.io/okta-mobile-swift/development/authfoundation/documentation/authfoundation/credential/authorize(_:)) async function, this will automatically refresh the credential's access token if it's expired, and add the appropriate HTTP authorization header to the request object. If you're not using Swift Concurrency, the same thing can be accomplished with the [`refreshIfNeeded(graceInterval:completion:)`](https://okta.github.io/okta-mobile-swift/development/authfoundation/documentation/authfoundation/credential/refreshifneeded(graceinterval:completion:)) and [`authorize(request:)`](https://okta.github.io/okta-mobile-swift/development/authfoundation/documentation/authfoundation/credential/authorize(request:)) functions.
```
added code tags 3 times